cpp/src/cider/common/Arena.h

```diff
@@ -40,35 +54,137 @@ namespace cider::hashtable {
  */
 class Arena : private boost::noncopyable {
```

Comment:
what's the relationship between `Arena`, `MemoryChunk`, `CiderXXAllocator`. From your description, `Arena` is an Allocator, while it didn't extend `CiderAllocator`, nor accept a customized Allocator. And if it's a allocator, please rename it as `ArenaAllocator`.
Besides, please create a cpp file to hide implementation.

Reply:
description editted